File: 4008-restore-finishing-order/restore-finishing-order.java
class Solution {
    public int[] recoverOrder(int[] order, int[] friends) {
        int[] arr = new int[friends.length];
        int index=0;
        for(int i=0;i<order.length;i++){
            for(int j=0;j<friends.length;j++){
                if(order[i]==friends[j]){
                    arr[index]=order[i];
                    index++;
                    break;
                }
            }
        }
        return arr;
    }
}

File: 4008-restore-finishing-order/restore-finishing-order.py
class Solution:
    def recoverOrder(self, order: List[int], friends: List[int]) -> List[int]:
        ans = [0] * len(friends)
        k=0
        for i in range(len(order)):
            for j in range(len(friends)):
                if order[i]==friends[j]:
                    ans[k]=order[i]
                    k+=1
                j+=1
            i+=1
        return ans

        